libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -686,16 +688,35 @@ class MapboxNavigation(
         tripSession.updateSensorEvent(sensorEvent)
     }
 
-    /**
-     * Updates the configuration to enable or disable the extended kalman filter (EKF).
-     *
-     * @param useEKF the new value for EKF
-     */
-    fun useExtendedKalmanFilter(useEKF: Boolean) {
-        tripSession.useExtendedKalmanFilter(useEKF)
+    private fun configureRouter() {
+        with(navigationOptions) {
+            ThreadController.getMainScopeAndRootJob().scope.launch {
```

Comment:
Why does this need to be on the main thread?

Reply:
yep, I just moved existing logic. We can change it here, not a problem, but maybe 
`navigator.configureRouter(routerParams)` should de called from the `main`?

Reply:
worth testing. am curious how long it would take, because there is no user interface. if it takes a long time, we should consider taking it off the ui thread - handle the synchronization issues separately

Reply:
Noting that this is causing synchronization issues as `setRoute` can be executed before `configureRouter` - `setRoute` is executed in IO and `configureRouter` in Main which causes this out of order chain of calls and therefore NN doesn't behave as expected. We noticed this when running Navigation UI SDK examples from the test app. Did some testing and changing to IO works but I'm wondering if somehow `setRoute` could be executed before `configureRouter` somehow as we have a dispatcher of 2 threads. It doesn't seem so but until `configureRouter` goes away https://github.com/mapbox/mapbox-navigation-android/pull/3395#discussion_r464945142 wondering if we should make sure that `configureRouter` is called always first and of course before `setRoute` differently.

Reply:
Follow up PR 👉 https://github.com/mapbox/mapbox-navigation-android/pull/3413